steam/webauth.py

```diff
@@ -325,6 +327,59 @@ def _finalize_login(self, login_response):
         self.steam_id = SteamID(data['steamid'])
         self.oauth_token = data['oauth_token']
 
+    def oauth_login(self, oauth_token='', steam_id='', language='english'):
+        if oauth_token:
+            self.oauth_token = oauth_token
+        else:
+            if self.oauth_token:
+                oauth_token = self.oauth_token
+            else:
+                raise LoginIncorrect('token is not specified')
+
+        if steam_id:
+            self.steam_id = SteamID(steam_id)
+        else:
+            if not self.steam_id:
+                raise LoginIncorrect('steam_id is not specified')
+
+        steam_id = self.steam_id.as_64
+
+        data = {
+            'access_token': oauth_token
+        }
+
+        try:
+            resp = self.session.post('https://api.steampowered.com/IMobileAuthService/GetWGToken/v0001', data=data)
+        except requests.exceptions.RequestException as e:
+            raise HTTPError(str(e))
+
+        try:
+            resp_data = resp.json()['response']
+        except json.decoder.JSONDecodeError as e:
+            if 'Please verify your <pre>key=</pre> parameter.' in resp.text:
+                raise LoginIncorrect('invalid token')
+            else:
+                raise e
```

Comment:
When there is issue with the response the raised exception should be the same with different message to simplify error handling.